src/gui/sdlmain.cpp

```diff
@@ -3107,8 +3125,14 @@ void Config_Add_SDL() {
 	pint->Set_help("Number of display to use; values depend on OS and user "
 	               "settings.");
 
-	Pbool = sdl_sec->Add_bool("vsync", deprecated, false);
-	Pbool->Set_help("Vertical sync setting not implemented (setting ignored)");
+	Pbool = sdl_sec->Add_bool("vsync", on_start, false);
+	Pbool->Set_help(
+	        "Sync to Vblank IF supported by the output device and renderer.\n"
+	        "It can reduce screen flickering, but it can also result in a slow DOSBox.");
+
+	pint = sdl_sec->Add_int("vsync_skip", on_start, 7000);
```

Comment:
> The 7000 feels too "under the hood" for a user setting.
> Like asking how many gigajoules of energy you'd like to buy at the petrol station.  I dunno.. fill the tank!
> milliseconds might be a little friendlier.
> ```ini
> vsync_skip_ms = 5
> ```
> Users wouldn't be able to go below 1 -> but from all our measurements on all our systems that's already far too low, and their experience will be bad anyway..so a lower-bound of 1 is probably the sane cutoff point.
> Also, whole milliseconds are pretty coarse, but I would imagine that there's little value in finagling at the sub-ms level anyway.  If someone needs to try 3.3 instead of 3, then we've probably got bigger problems.

Is it really that different, conceptually, from `[mixer] blocksize = 2048`? I dunno... big enough! Should we just use kilobytes?

> Thinking about an upper-bound:
> Anything larger than the audio buffer will guarantee stuttering - so this is a fail-zone. It can be calculated using the prebuffer ms + buffer-block size->ms calculation, but realistically this is around 32 to 64ms.
> Perhaps a better upper-bound is the maximum vsync tyime itself, typically (16.6ms @ 60hz).
> But maybe there's still some processing overhead - so maybe we make it 2 x frame-period ... or even 10x-frame-period?
> sdlmain has this value:
> ```c++
> frame_period = std::chrono::nanoseconds(1'000'000'000 / refresh_rate);
> ```
> This variable is kept up-to-date on every video mode change; so will get bigger if the user  switches to a slow refresh rate; say 24 Hz.
> ```

For an upper bound, can we get by with another hard-coded value, say 12000? The goal of this pacing is to leave some remainder of the ~14ms 70Hz rate for other processing. Hopefully we can avoid trying to get too clever? I get no stuttering with `vsync_skip = 12000` and `blocksize = 1024` `prebuffer=10`. Maybe just a warning in the console for people who are trying to get crazy?

Reply:
> Is it really that different, conceptually, from [mixer] blocksize = 2048? I dunno... big enough! Should we just use kilobytes?

Thanks for this comparison! - I now see the two (µs vs ms) as not being any different.

Reply:
> For an upper bound, can we get by with another hard-coded value, say 12000

:+1: 

That sounds perfectly reasonable - and is much easier to document why 12ms was selected too.